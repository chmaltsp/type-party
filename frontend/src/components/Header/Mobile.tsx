import * as React from 'react';

import { em } from 'polished';
import styled, { media } from 'sc';

import LinkBase from '../Link';

const Link = styled(LinkBase)`
  width: 100%;
  padding: ${em(16)} 0;
  text-align: center;
  text-decoration: none;
`;

import { links } from './';

const LinkWrapper = styled.ul<{ open: boolean }>`
  display: none;
  flex-direction: column;
  padding: 0;
  margin: 0;
  max-height: ${({ open }) => (open ? em(links.length * 50) : '0')};
  overflow: hidden;
  transition: max-height 0.2s linear;
  ${media.tablet`
    display: flex;
  `};
`;

export interface MobileNavProps {
  open: boolean;
}

const MobileNav: React.SFC<MobileNavProps> = ({ open }) => (
  <LinkWrapper open={open}>
    {links.map(({ to, text }) => {
      return (
        <Link key={to} to={to}>
          {text}
        </Link>
      );
    })}
  </LinkWrapper>
);

export default MobileNav;
