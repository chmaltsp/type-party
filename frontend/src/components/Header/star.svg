<svg width="50" height="50" viewBox="0 0 50 50" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M24.419 37.5319L13.5515 47.5646L14.6277 32.8215L2.88124e-06 30.589L12.2095 22.2374L4.83648 9.42075L18.9853 13.7496L24.419 1.19388e-07L29.8528 13.7496L44.0016 9.42075L36.6285 22.2374L48.838 30.589L34.2103 32.8215L35.2865 47.5646L24.419 37.5319Z" transform="matrix(0.939851 -0.341585 -0.342455 -0.939534 11.2888 57.3709)" fill="#F8D022"/>
</svg>
