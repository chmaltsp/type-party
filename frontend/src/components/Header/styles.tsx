import { em, transparentize } from 'polished';

import styled, { media } from 'sc';

import { Star as StarBase } from '../Images';

import LinkBase from '../Link';
import LogoBase from '../Logo';

const Link = styled(LinkBase)`
  font-size: ${em(18)};
  font-weight: 400;
  text-decoration: none;
  margin-right: ${({ theme }) => em(theme.spacing.md)};
`;

const Wrapper = styled.div`
  display: flex;
  margin: 0 ${em(136)};
  padding-top: ${em(48)};
  padding-bottom: ${em(16)};
  border-bottom: none;
  ${media.tablet`
  border-bottom: 1px solid ${(props: any) =>
    transparentize(0.8, props.theme.colors.black)};
`};
  ${media.desktop`
  margin: 0 ${em(32)};
`};
`;

const Star = styled(StarBase)`
  display: none;
  width: 48px;
  height: 48px;
  background-repeat: no-repeat;
  ${media.tablet`
    display: inline-block;
  `};
`;

const Logo = styled(LogoBase)`
  display: block;
  width: ${em(80)};
  ${media.tablet`
    display: none;
  `}
  ${media.desktop`
    width: ${em(72)};
    height: ${em(72)};
  `};
`;

const NavlinksWrapper = styled.div`
  flex: 1;
  text-align: right;
  ${media.tablet`
      display: none;
  `};
`;

const MenuLink = styled.div`
  font-size: ${em(18)};
  font-weight: 400;
  margin-left: auto;
  display: none;
  align-self: center;
  cursor: pointer;
  ${media.tablet`
      display: inline-block;
  `};
`;

export { MenuLink, NavlinksWrapper, Logo, Star, Wrapper, Link };
