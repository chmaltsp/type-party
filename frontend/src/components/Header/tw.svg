<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M30.6297 3.61193C30.6297 3.61193 28.9646 4.95475 26.628 5.19646C26.628 5.19646 22.9487 1.08737 17.6847 4.65934C15.3213 6.32444 15.1065 9.84272 15.5093 11.105C15.5093 11.105 8.36545 11.669 2.13466 4.36394C2.13466 4.36394 -0.65842 9.19815 3.9609 13.0655C3.9609 13.0655 2.59559 13.2804 1.29776 12.2061C1.29776 12.2061 0.684332 17.2015 6.27056 18.5981C6.27056 18.5981 5.89459 19.2426 3.63862 18.5981C3.63862 18.5981 4.01459 22.6803 9.33223 23.2711C9.33223 23.2711 6.45948 26.2254 0 25.8494C0 25.8494 10.6751 33.8527 22.2235 24.1843C28.6691 18.7055 28.4543 10.8096 28.3468 9.62795C28.3468 9.62795 30.8923 7.69426 31.5533 6.45883C31.5533 6.45883 29.5822 7.10339 27.9708 7.10339C27.9708 7.10326 30.1194 5.97528 30.6297 3.61193Z" fill="#61B2E4"/>
</svg>
