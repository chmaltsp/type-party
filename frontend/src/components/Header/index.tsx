import * as React from 'react';

import MobileNav from './Mobile';

import {
  Line,
  Link,
  LinkA,
  Logo,
  MenuLink,
  NavlinksWrapper,
  Star,
  TwitterLogo,
  Wrapper,
} from './styles';

import { Link as RouterLink } from 'react-router-dom';

import tw from './tw.svg';

interface NavLink {
  to: string;
  text: string | React.ReactNode;
}
type NavLinks = NavLink[];

export const links: NavLinks = [
  {
    text: 'Font Directory',
    to: '/directory',
  },
  {
    text: 'Sites using free fonts',
    to: '/',
  },
  {
    text: 'About',
    to: '/about',
  },
];

export interface HeaderState {
  mobileOpen: boolean;
}

class Header extends React.Component<{}, HeaderState> {
  constructor(props: {}) {
    super(props);

    this.state = {
      mobileOpen: false,
    };
  }

  public handleMenuToggle = () => {
    if (this.state.mobileOpen) {
      this.setState({
        mobileOpen: false,
      });
    } else {
      this.setState({
        mobileOpen: true,
      });
    }
  };
  public render() {
    return (
      <React.Fragment>
        <Wrapper>
          <RouterLink to="/">
            <Logo />
          </RouterLink>
          <Star />
          <MenuLink onClick={this.handleMenuToggle}>Menu</MenuLink>
          <NavlinksWrapper>
            <>
              {links.map(({ to, text }) => (
                <Link key={to} to={to}>
                  {text}
                </Link>
              ))}
              <LinkA href="https://twitter.com/typeparty_" target="_blank">
                <TwitterLogo src={tw} alt="Twitter Type Party" />
              </LinkA>
            </>
          </NavlinksWrapper>
        </Wrapper>
        <Line />
        <MobileNav open={this.state.mobileOpen} />
      </React.Fragment>
    );
  }
}

export default Header;
