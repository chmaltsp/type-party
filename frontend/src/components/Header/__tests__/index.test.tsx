import * as React from 'react';

import Header, { HeaderState } from '../';

import { shallow } from 'enzyme';

describe('<Header />', () => {
  const component = shallow<{}, HeaderState>(<Header />);
  it('should render without blowing up', () => {
    expect(component).toMatchSnapshot();
  });

  it('should open and close the mobile menu', () => {
    const instance = component.instance();
    // @ts-ignore
    instance.handleMenuToggle();

    expect((instance.state as any).mobileOpen).toEqual(true);
    // @ts-ignore
    instance.handleMenuToggle();

    expect((instance.state as any).mobileOpen).toEqual(false);
  });
});
