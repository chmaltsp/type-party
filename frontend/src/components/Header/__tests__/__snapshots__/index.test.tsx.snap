// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render without blowing up 1`] = `
<Fragment>
  <styled.div>
    <styled.img />
    <styled.div />
    <styled.div
      onClick={[Function]}
    >
      Menu
    </styled.div>
    <styled.div>
      <Styled(Styled(Link))
        key="/directory"
        to="/directory"
      >
        Font Directory
      </Styled(Styled(Link))>
      <Styled(Styled(Link))
        key="/free-fonts"
        to="/free-fonts"
      >
        Sites using free fonts
      </Styled(Styled(Link))>
      <Styled(Styled(Link))
        key="/resources"
        to="/resources"
      >
        Resources
      </Styled(Styled(Link))>
      <Styled(Styled(Link))
        key="/about"
        to="/about"
      >
        About
      </Styled(Styled(Link))>
      <Styled(Styled(Link))
        key="/twitter"
        to="/twitter"
      >
        <span>
          @typeparty
        </span>
      </Styled(Styled(Link))>
    </styled.div>
  </styled.div>
  <MobileNav
    open={false}
  />
</Fragment>
`;
